provider/ec2/environ.go

```diff
@@ -683,6 +683,27 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	}, nil
 }
 
+// FilterSubnetsToZones removes the INFAN network types from possible subnets.
+// We can't use the INFAN networks as they require you to use the underlay
+// network.
+// TODO (stickupkid): We should lift this to core package if more providers need
+// this.
+func filterSubnetsToZones(subnetsToZones []map[corenetwork.Id][]string) map[corenetwork.Id][]string {
```

Comment:
The note I would make here is about potentially omitting them during generation of `ProvisioningInfo` provided that container provisioning was unaffected.

Reply:
Ah good shout.